[agent builder] Update models docs for 9.4 model management UI (#6355)
elastic/docs-content: explore-analyze/ai-features/agent-builder/glossary.md

## G

$$$generative-ai-connector$$$
Generative AI connector

> :   A {{kib}} connector that wires {{agent-builder}} to an LLM provider such as OpenAI, Anthropic, Amazon Bedrock, Google Gemini, or Azure OpenAI. Distinct from a [](#connector), which connects {{agent-builder}} to non-LLM external services. See [](models.md#configure-a-connector).

Reviewer: ```suggestion
:   A {{kib}} connector that connects {{agent-builder}} to an LLM provider such as OpenAI, Anthropic, Amazon Bedrock, Google Gemini, or Azure OpenAI. Distinct from a [](#connector), which connects {{agent-builder}} to non-LLM external services. See [](models.md#configure-a-connector).
```

language nit

Index search tool
:   A type of custom tool that performs natural-language search over a configured set of indices, aliases, or data streams. The tool selects an appropriate query strategy (keyword, semantic, or hybrid) automatically. See [](tools/index-search-tools.md#when-to-use-index-search-tools).

$$$inference-endpoint$$$
Inference endpoint

> :   An {{es}} resource that wires the cluster to a model: Elastic-managed (through EIS) or third-party (OpenAI, Anthropic, Amazon Bedrock, Cohere, Google AI, and others). {{agent-builder}} can use any inference endpoint that supports the `chat_completion` task type as a model source. Inference endpoints are managed from **Elastic inference**, **External inference**, or the [{{infer}} APIs]({{es-apis}}group/endpoint-inference). See [](models.md#add-an-inference-endpoint).

Reviewer: ```suggestion
:   An {{es}} resource that connects the cluster to a third-party or Elastic-managed model on EIS. {{agent-builder}} can use any inference endpoint that supports the `chat_completion` task type as a model source. Inference endpoints are managed from **Elastic inference**, **External inference**, or the [{{infer}} APIs]({{es-apis}}group/endpoint-inference). See [](models.md#add-an-inference-endpoint).
```

$$$model$$$
Model

> :   The LLM that an agent uses to reason and produce responses. {{agent-builder}} can use Elastic Managed LLMs, third-party models reached through an [](#inference-endpoint), or models reached through a [](#generative-ai-connector). See [](models.md#use-additional-models).

Reviewer: there's a better verb than "reached" here
